sound/soc/sof/topology.c

```diff
@@ -2161,6 +2161,12 @@ static int sof_link_hda_process(struct snd_sof_dev *sdev,
 
 	config->hda.link_dma_ch = slot;
 
+	/* Set the DAI config here since both DAI name and direction must
+	 * be matched, this is important for playback and capture with the
+	 * same DAI name.
```

Comment:
I couldn't figure out what the comment means. You cannot match the DAI direction for playback and capture, the two parts of the sentence are logically opposite.

Reply:
I mean if the playback and capture have the same DAI name, we need both check the name and the direction to find the right dai we need, and thus set the dai config.